### PR TITLE
Conditional UI

### PR DESCRIPTION
**Pull request by snehmehta**
Ticket: #330

```yaml
- Conditional:
    conditions:
      - if: ${1 != 1}
        then:
          Text:
            text: If condition works
      - elif: ${1 != 1}
        then:
          Text:
            text: First elif condition works
      - elif: ${1 == 1}
        then:
          Text:
            text: Second elif condition works
      - else:
          Text:
            text: oops no match

```

### CODE REVIEW

**Comment by kmahmood74** [comment]
hi @snehmehta I think we should stick to what everyone knows - elseif and not invent elif. Already people have to learn Ensemble, we must not add more cognitive burden when this works exactly like if/then/else. elseif is a well know thing that people understand.

**Comment by kmahmood74** [comment]
@snehmehta sorry I may take that back. I think you are inspired by python's elif. that is quite relevant for us as python uses indentation and doesn't have braces. 

PHP uses elseif

@vusters @evshi what do you think? I am fine with elif as well.

File: lib/widget/conditional.dart [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    try {
+      return scopeManager != null &&
+          conditions != null &&
+          (conditions is YamlList) &&
+          conditions.isNotEmpty &&
+          conditions.first.containsKey('if') &&
+          conditions.first.containsKey('then');
+    } on Exception catch (_) {
+      throw LanguageError(
+          'In Proper structure, make sure atleast if then condition ');
```
Comment (evshi) [inline lib/widget/conditional.dart:83]: In proper -> Improper

**Review by evshi — COMMENTED** [review]
`elif` looks good to me. Is there any value in having the `then`?

File: lib/widget/conditional.dart (hunk 1 above)
Comment (snehmehta) [reply lib/widget/conditional.dart:83]: Thanks, fixed.

**Comment by snehmehta** [comment]
@evshi `then` mainly to pick widget definition, also readability if true then build this widget.

**Comment by vusters** [comment]
elseif, elsif, elif - personally like the first option - read best for non developer. Plus in the Property Panel we probably will spell out 'else if', so without space in the Code editor is the closest to it.

**Comment by snehmehta** [comment]
> You need to handle binding changes (watch for all changes to all of the expressions, then re-execute the entire if/elseif/then). See Scope's listen() method. It essentially want a setter on this Conditional widget that it can call when bindings have changed.

thanks! We'll do the changes.

**Comment by snehmehta** [comment]
Updated example with binding changes.

```yaml
- Button:
    label: Capture
    onTap:
      openCamera:
        id: cameraId
        options:
          mode: photo

- Conditional:
    conditions:
      - if: ${cameraId.files.length == 1}
        Text:
          text: Single Image
      - elseif: ${cameraId.files.length == 2}
        Text:
          text: Double Image
      - elseif: ${cameraId.files.length >= 3}
        Text:
          text: multiple image
      - else:
        Text:
          text: No images

```

Also, agreed with @vusters elseif is better in readability than elif.

**Review by kmahmood74 — APPROVED** [review]

**Comment by kmahmood74** [comment]
@snehmehta please merge

**Pull request merged**